### PR TITLE
[DeviceSanitizerLayer] Demangle the kernel name in error reporting

### PR DESCRIPTION
**Pull request by yingcong-wu**
Try to demangle the kernel name in error reporting to improve user experience.

### CODE REVIEW

File: source/loader/layers/sanitizer/asan_interceptor.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 #include "asan_interceptor.hpp"
 #include "ur_sanitizer_layer.hpp"
+#include <cxxabi.h>
```
Comment (AllanZyne) [inline source/loader/layers/sanitizer/asan_interceptor.cpp:16]: IMO, it'd better to extract these codes as a function, which is defined in `unified-runtime/source/loader/layers/sanitizer/linux/san_utils.cpp`, as it's specific to Linux

**Review by AllanZyne — COMMENTED** [review]
IMO, it'd better to extract these codes as a function, which is defined in `unified-runtime/source/loader/layers/sanitizer/linux/san_utils.cpp`, as it's specific to Linux

**Comment by yingcong-wu** [comment]
> IMO, it'd better to extract these codes as a function, which is defined in `unified-runtime/source/loader/layers/sanitizer/linux/san_utils.cpp`, as it's specific to Linux

Updated.

File: source/loader/layers/sanitizer/linux/san_utils.cpp [hunk 2]
```diff
@@ -84,4 +86,15 @@ void *GetMemFunctionPointer(const char *FuncName) {
...
+std::string DemangleName(const std::string &name) {
+    std::string result = name;
+    char *demangled =
+        abi::__cxa_demangle(name.c_str(), nullptr, nullptr, nullptr);
+    if (demangled) {
+        result = demangled;
+        free(demangled);
+    }
+    return result;
+}
```
Comment (AllanZyne) [inline source/loader/layers/sanitizer/linux/san_utils.cpp:98]: If "DemangleName" failed, the "KernelName" will be an empty string. 
How about "void DemangleName(std::string& name)"? We aren't interested in whether demangling succeeds or fails anyway.

Comment (yingcong-wu) [reply source/loader/layers/sanitizer/linux/san_utils.cpp:98]: `std::string result = name;` makes sure if demangling failed, then the name is unchanged.

Comment (AllanZyne) [reply source/loader/layers/sanitizer/linux/san_utils.cpp:98]: Okay. Does "abi::__cxa_demangle" need to link a library?

**Review by AllanZyne — APPROVED** [review]

**Review by AllanZyne — APPROVED** [review]
LGTM

**Pull request merged**